rdrf/rdrf/admin.py

```diff
@@ -129,93 +161,133 @@ def export_registry(registry, request):
 
+                messages.success(request, _(f"Registration enabled for {registry}"))
 
-generate_questionnaire_action.short_description = _("Generate Questionnaire")
+            notifications = EmailNotification.objects.filter(
+                registry=registry,
+                description__in=EventType.REGISTRATION_TYPES,
+            )
+            all_notifications_disabled = not notifications.filter(disabled=False).exists()
 
+            if not notifications.exists():
+                messages.warning(request, render_to_string("admin/notifications_needed.html", {
+                    "registry": registry
+                }))
+            elif all_notifications_disabled:
+                messages.warning(request, render_to_string("admin/notifications_disabled.html", {
+                    "registry": registry
+                }))
 
-class RegistryAdmin(admin.ModelAdmin):
-    actions = [export_registry_action, generate_questionnaire_action]
+        if unchanged:
+            messages.info(request, _(f"'{', '.join(desc for desc in unchanged)}' already enabled registration"))
 
-    def get_queryset(self, request):
-        if not request.user.is_superuser:
-            user = get_user_model().objects.get(username=request.user)
-            return Registry.objects.filter(registry__in=[reg.id for reg in user.registry.all()])
+    enable_registration_action.short_description = _("Enable registration")
 
-        return Registry.objects.all()
+    def disable_registration_action(self, request, registry_models_selected):
+        unchanged = []
 
-    def has_add_permission(self, request, *args, **kwargs):
-        if request.user.is_superuser:
-            return True
-        return False
+        for registry in registry_models_selected:
+            if not registry.has_feature(RegistryFeatures.REGISTRATION):
+                unchanged.append(str(registry))
+            else:
+                registry.remove_feature(RegistryFeatures.REGISTRATION)
+                registry.save()
 
-    def has_change_permission(self, request, *args, **kwargs):
-        if request.user.is_superuser:
-            return True
-        return False
+                messages.success(request, _(f"Registration disabled for {registry}"))
 
-    def has_delete_permission(self, request, *args, **kwargs):
-        if request.user.is_superuser:
-            return True
-        return False
+        if unchanged:
+            messages.info(request, _(f"'{', '.join(desc for desc in unchanged)}' already had registration disabled"))
 
-    def get_urls(self):
-        original_urls = super(RegistryAdmin, self).get_urls()
+    disable_registration_action.short_description = _("Disable registration")
 
-        return original_urls
+    def create_notifications_action(self, request, registry_models_selected):
+        for registry in registry_models_selected:
+            # TODO: Add default notifications for all registration types. Needs different recipients.
```

Comment:
Shouldn't we check that there are no notifications (of the same type) set up already here and don't add any if so?

Reply:
Well the existing notifications might be for the same event, but are being sent to clinicians rather than patients.